pymc3/distributions/posterior_predictive.py

```diff
@@ -50,21 +52,14 @@
 #    test_mixture_random_shape::test_mixture_random_shape
 #
 
-PosteriorPredictiveTrace = Dict[str, np.ndarray]
 Point = Dict[str, np.ndarray]
 
 
 class HasName(Protocol):
     name: str
 
 
-if TYPE_CHECKING:
-    _TraceDictParent = UserDict[str, np.ndarray]
-else:
-    _TraceDictParent = UserDict
```

Comment:
this strikes me as unnecessarily complicated, we can just annotate `data` directly in `_TraceDict`